<!DOCTYPE html>

<html lang="en">

<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=no">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black">
    <meta name="mobile-web-app-capable" content="yes">
    <title>
        Genes of Mice and Men - HackMD
    </title>
    <link rel="icon" type="image/png" href="https://hackmd.io/favicon.png">
    <link rel="apple-touch-icon" href="https://hackmd.io/apple-touch-icon.png">

    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/ionicons/2.0.1/css/ionicons.min.css" integrity="sha256-3iu9jgsy9TpTwXKb7bNQzqWekRX7pPK+2OLj3R922fo=" crossorigin="anonymous" />
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/octicons/3.5.0/octicons.min.css" integrity="sha256-QiWfLIsCT02Sdwkogf6YMiQlj4NE84MKkzEMkZnMGdg=" crossorigin="anonymous" />
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.5.1/themes/prism.min.css" integrity="sha256-vtR0hSWRc3Tb26iuN2oZHt3KRUomwTufNIf5/4oeCyg=" crossorigin="anonymous" />
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@hackmd/emojify.js@2.1.0/dist/css/basic/emojify.min.css" integrity="sha256-UOrvMOsSDSrW6szVLe8ZDZezBxh5IoIfgTwdNDgTjiU=" crossorigin="anonymous" />
    <style>
        @import url(https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,500,500i|Source+Code+Pro:300,400,500|Source+Sans+Pro:300,300i,400,400i,600,600i|Source+Serif+Pro&subset=latin-ext);.hljs{background:#fff;color:#333;display:block;overflow-x:auto;padding:.5em}.hljs-comment,.hljs-meta{color:#969896}.hljs-emphasis,.hljs-quote,.hljs-string,.hljs-strong,.hljs-template-variable,.hljs-variable{color:#df5000}.hljs-keyword,.hljs-selector-tag,.hljs-type{color:#a71d5d}.hljs-attribute,.hljs-bullet,.hljs-literal,.hljs-number,.hljs-symbol{color:#0086b3}.hljs-built_in,.hljs-builtin-name{color:#005cc5}.hljs-name,.hljs-section{color:#63a35c}.hljs-tag{color:#333}.hljs-attr,.hljs-selector-attr,.hljs-selector-class,.hljs-selector-id,.hljs-selector-pseudo,.hljs-title{color:#795da3}.hljs-addition{background-color:#eaffea;color:#55a532}.hljs-deletion{background-color:#ffecec;color:#bd2c00}.hljs-link{text-decoration:underline}.markdown-body{word-wrap:break-word;font-size:16px;line-height:1.5}.markdown-body:after,.markdown-body:before{content:"";display:table}.markdown-body:after{clear:both}.markdown-body>:first-child{margin-top:0!important}.markdown-body>:last-child{margin-bottom:0!important}.markdown-body a:not([href]){color:inherit;text-decoration:none}.markdown-body .absent{color:#c00}.markdown-body .anchor{float:left;line-height:1;margin-left:-20px;padding-right:4px}.markdown-body .anchor:focus{outline:none}.markdown-body blockquote,.markdown-body dl,.markdown-body ol,.markdown-body p,.markdown-body pre,.markdown-body table,.markdown-body ul{margin-bottom:16px;margin-top:0}.markdown-body hr{background-color:#e7e7e7;border:0;height:.25em;margin:24px 0;padding:0}.markdown-body blockquote{border-left:.25em solid #ddd;color:#777;font-size:16px;padding:0 1em}.markdown-body blockquote>:first-child{margin-top:0}.markdown-body blockquote>:last-child{margin-bottom:0}.markdown-body kbd,.popover kbd{background-color:#fcfcfc;border:1px solid;border-color:#ccc #ccc #bbb;border-radius:3px;box-shadow:inset 0 -1px 0 #bbb;color:#555;display:inline-block;font-size:11px;line-height:10px;padding:3px 5px;vertical-align:middle}.markdown-body .loweralpha{list-style-type:lower-alpha}.markdown-body h1,.markdown-body h2,.markdown-body h3,.markdown-body h4,.markdown-body h5,.markdown-body h6{font-weight:600;line-height:1.25;margin-bottom:16px;margin-top:24px}.markdown-body h1 .octicon-link,.markdown-body h2 .octicon-link,.markdown-body h3 .octicon-link,.markdown-body h4 .octicon-link,.markdown-body h5 .octicon-link,.markdown-body h6 .octicon-link{color:#000;vertical-align:middle;visibility:hidden}.markdown-body h1:hover .anchor,.markdown-body h2:hover .anchor,.markdown-body h3:hover .anchor,.markdown-body h4:hover .anchor,.markdown-body h5:hover .anchor,.markdown-body h6:hover .anchor{text-decoration:none}.markdown-body h1:hover .anchor .octicon-link,.markdown-body h2:hover .anchor .octicon-link,.markdown-body h3:hover .anchor .octicon-link,.markdown-body h4:hover .anchor .octicon-link,.markdown-body h5:hover .anchor .octicon-link,.markdown-body h6:hover .anchor .octicon-link{visibility:visible}.markdown-body h1 code,.markdown-body h1 tt,.markdown-body h2 code,.markdown-body h2 tt,.markdown-body h3 code,.markdown-body h3 tt,.markdown-body h4 code,.markdown-body h4 tt,.markdown-body h5 code,.markdown-body h5 tt,.markdown-body h6 code,.markdown-body h6 tt{font-size:inherit}.markdown-body h1{font-size:2em}.markdown-body h1,.markdown-body h2{border-bottom:1px solid #eee;padding-bottom:.3em}.markdown-body h2{font-size:1.5em}.markdown-body h3{font-size:1.25em}.markdown-body h4{font-size:1em}.markdown-body h5{font-size:.875em}.markdown-body h6{color:#777;font-size:.85em}.markdown-body ol,.markdown-body ul{padding-left:2em}.markdown-body ol.no-list,.markdown-body ul.no-list{list-style-type:none;padding:0}.markdown-body ol ol,.markdown-body ol ul,.markdown-body ul ol,.markdown-body ul ul{margin-bottom:0;margin-top:0}.markdown-body li>p{margin-top:16px}.markdown-body li+li{padding-top:.25em}.markdown-body dl{padding:0}.markdown-body dl dt{font-size:1em;font-style:italic;font-weight:700;margin-top:16px;padding:0}.markdown-body dl dd{margin-bottom:16px;padding:0 16px}.markdown-body table{display:block;overflow:auto;width:100%;word-break:normal;word-break:keep-all}.markdown-body table th{font-weight:700}.markdown-body table td,.markdown-body table th{border:1px solid #ddd;padding:6px 13px}.markdown-body table tr{background-color:#fff;border-top:1px solid #ccc}.markdown-body table tr:nth-child(2n){background-color:#f8f8f8}.markdown-body img{background-color:#fff;box-sizing:initial;max-width:100%}.markdown-body img[align=right]{padding-left:20px}.markdown-body img[align=left]{padding-right:20px}.markdown-body .emoji{background-color:initial;max-width:none;vertical-align:text-top}.markdown-body span.frame{display:block;overflow:hidden}.markdown-body span.frame>span{border:1px solid #ddd;display:block;float:left;margin:13px 0 0;overflow:hidden;padding:7px;width:auto}.markdown-body span.frame span img{display:block;float:left}.markdown-body span.frame span span{clear:both;color:#333;display:block;padding:5px 0 0}.markdown-body span.align-center{clear:both;display:block;overflow:hidden}.markdown-body span.align-center>span{display:block;margin:13px auto 0;overflow:hidden;text-align:center}.markdown-body span.align-center span img{margin:0 auto;text-align:center}.markdown-body span.align-right{clear:both;display:block;overflow:hidden}.markdown-body span.align-right>span{display:block;margin:13px 0 0;overflow:hidden;text-align:right}.markdown-body span.align-right span img{margin:0;text-align:right}.markdown-body span.float-left{display:block;float:left;margin-right:13px;overflow:hidden}.markdown-body span.float-left span{margin:13px 0 0}.markdown-body span.float-right{display:block;float:right;margin-left:13px;overflow:hidden}.markdown-body span.float-right>span{display:block;margin:13px auto 0;overflow:hidden;text-align:right}.markdown-body code,.markdown-body tt{background-color:#0000000a;border-radius:3px;font-size:85%;margin:0;padding:.2em 0}.markdown-body code:after,.markdown-body code:before,.markdown-body tt:after,.markdown-body tt:before{content:"\00a0";letter-spacing:-.2em}.markdown-body code br,.markdown-body tt br{display:none}.markdown-body del code{text-decoration:inherit}.markdown-body pre{word-wrap:normal}.markdown-body pre>code{background:#0000;border:0;font-size:100%;margin:0;padding:0;white-space:pre;word-break:normal}.markdown-body .highlight{margin-bottom:16px}.markdown-body .highlight pre{margin-bottom:0;word-break:normal}.markdown-body .highlight pre,.markdown-body pre{background-color:#f7f7f7;border-radius:3px;font-size:85%;line-height:1.45;overflow:auto;padding:16px}.markdown-body pre code,.markdown-body pre tt{word-wrap:normal;background-color:initial;border:0;display:inline;line-height:inherit;margin:0;max-width:auto;overflow:visible;padding:0}.markdown-body pre code:after,.markdown-body pre code:before,.markdown-body pre tt:after,.markdown-body pre tt:before{content:normal}.markdown-body .csv-data td,.markdown-body .csv-data th{font-size:12px;line-height:1;overflow:hidden;padding:5px;text-align:left;white-space:nowrap}.markdown-body .csv-data .blob-line-num{background:#fff;border:0;padding:10px 8px 9px;text-align:right}.markdown-body .csv-data tr{border-top:0}.markdown-body .csv-data th{background:#f8f8f8;border-top:0;font-weight:700}.news .alert .markdown-body blockquote{border:0;padding:0 0 0 40px}.activity-tab .news .alert .commits,.activity-tab .news .markdown-body blockquote{padding-left:0}.task-list-item{list-style-type:none}.task-list-item label{font-weight:400}.task-list-item.enabled label{cursor:pointer}.task-list-item+.task-list-item{margin-top:3px}.task-list-item-checkbox{cursor:default!important;float:left;margin:.31em 0 .2em -1.3em!important;vertical-align:middle}.markdown-body{max-width:758px;overflow:visible!important;padding-bottom:40px;padding-top:40px;position:relative}.markdown-body .emoji{vertical-align:top}.markdown-body pre{border:inherit!important}.markdown-body code{color:inherit!important}.markdown-body pre code .wrapper{display:-moz-inline-flex;display:-ms-inline-flex;display:-o-inline-flex;display:inline-flex}.markdown-body pre code .gutter{float:left;overflow:hidden;-webkit-user-select:none;user-select:none}.markdown-body pre code .gutter.linenumber{border-right:3px solid #6ce26c!important;box-sizing:initial;color:#afafaf!important;cursor:default;display:inline-block;min-width:20px;padding:0 8px 0 0;position:relative;text-align:right;z-index:4}.markdown-body pre code .gutter.linenumber>span:before{content:attr(data-linenumber)}.markdown-body pre code .code{float:left;margin:0 0 0 16px}.markdown-body .gist .line-numbers{border-bottom:none;border-left:none;border-top:none}.markdown-body .gist .line-data{border:none}.markdown-body .gist table{border-collapse:inherit!important;border-spacing:0}.markdown-body code[data-gist-id]{background:none;padding:0}.markdown-body code[data-gist-id]:after,.markdown-body code[data-gist-id]:before{content:""}.markdown-body code[data-gist-id] .blob-num{border:unset}.markdown-body code[data-gist-id] table{margin-bottom:unset;overflow:unset}.markdown-body code[data-gist-id] table tr{background:unset}.markdown-body[dir=rtl] pre{direction:ltr}.markdown-body[dir=rtl] code{direction:ltr;unicode-bidi:embed}.markdown-body .alert>p:last-child{margin-bottom:0}.markdown-body pre.abc,.markdown-body pre.flow-chart,.markdown-body pre.graphviz,.markdown-body pre.mermaid,.markdown-body pre.sequence-diagram,.markdown-body pre.vega{background-color:inherit;border-radius:0;overflow:visible;text-align:center;white-space:inherit}.markdown-body pre.abc>code,.markdown-body pre.flow-chart>code,.markdown-body pre.graphviz>code,.markdown-body pre.mermaid>code,.markdown-body pre.sequence-diagram>code,.markdown-body pre.vega>code{text-align:left}.markdown-body pre.abc>svg,.markdown-body pre.flow-chart>svg,.markdown-body pre.graphviz>svg,.markdown-body pre.mermaid>svg,.markdown-body pre.sequence-diagram>svg,.markdown-body pre.vega>svg{height:100%;max-width:100%}.markdown-body pre>code.wrap{word-wrap:break-word;white-space:pre-wrap;white-space:-moz-pre-wrap;white-space:-pre-wrap;white-space:-o-pre-wrap}.markdown-body .alert>p:last-child,.markdown-body .alert>ul:last-child{margin-bottom:0}.markdown-body summary{display:list-item}.markdown-body summary:focus{outline:none}.markdown-body details summary{cursor:pointer}.markdown-body details:not([open])>:not(summary){display:none}.markdown-body figure{margin:1em 40px}.markdown-body .mark,.markdown-body mark{background-color:#fff1a7}.vimeo,.youtube{background-color:#000;background-position:50%;background-repeat:no-repeat;background-size:contain;cursor:pointer;display:table;overflow:hidden;text-align:center}.vimeo,.youtube{position:relative;width:100%}.youtube{padding-bottom:56.25%}.vimeo img{object-fit:contain;width:100%;z-index:0}.youtube img{object-fit:cover;z-index:0}.vimeo iframe,.youtube iframe,.youtube img{height:100%;left:0;position:absolute;top:0;width:100%}.vimeo iframe,.youtube iframe{vertical-align:middle;z-index:1}.vimeo .icon,.youtube .icon{color:#fff;height:auto;left:50%;opacity:.3;position:absolute;top:50%;transform:translate(-50%,-50%);transition:opacity .2s;width:auto;z-index:0}.vimeo:hover .icon,.youtube:hover .icon{opacity:.6;transition:opacity .2s}.slideshare .inner,.speakerdeck .inner{position:relative;width:100%}.slideshare .inner iframe,.speakerdeck .inner iframe{bottom:0;height:100%;left:0;position:absolute;right:0;top:0;width:100%}.figma{display:table;padding-bottom:56.25%;position:relative;width:100%}.figma iframe{border:1px solid #eee;bottom:0;height:100%;left:0;position:absolute;right:0;top:0;width:100%}.markmap-container{height:300px}.markmap-container>svg{height:100%;width:100%}.MJX_Assistive_MathML{display:none}#MathJax_Message{z-index:1000!important}.ui-infobar{color:#777;margin:25px auto -25px;max-width:760px;position:relative;z-index:2}.toc .invisable-node{list-style-type:none}.ui-toc{bottom:20px;position:fixed;z-index:998}.ui-toc.both-mode{margin-left:8px}.ui-toc.both-mode .ui-toc-label{border-bottom-left-radius:0;border-top-left-radius:0;height:40px;padding:10px 4px}.ui-toc-label{background-color:#e6e6e6;border:none;color:#868686;transition:opacity .2s}.ui-toc .open .ui-toc-label{color:#fff;opacity:1;transition:opacity .2s}.ui-toc-label:focus{background-color:#ccc;color:#000;opacity:.3}.ui-toc-label:hover{background-color:#ccc;opacity:1;transition:opacity .2s}.ui-toc-dropdown{margin-bottom:20px;margin-top:20px;max-height:70vh;max-width:45vw;overflow:auto;padding-left:10px;padding-right:10px;text-align:inherit;width:25vw}.ui-toc-dropdown>.toc{max-height:calc(70vh - 100px);overflow:auto}.ui-toc-dropdown[dir=rtl] .nav{letter-spacing:.0029em;padding-right:0}.ui-toc-dropdown a{overflow:hidden;text-overflow:ellipsis;white-space:pre}.ui-toc-dropdown .nav>li>a{color:#767676;display:block;font-size:13px;font-weight:500;padding:4px 20px}.ui-toc-dropdown .nav>li:first-child:last-child>ul,.ui-toc-dropdown .toc.expand ul{display:block}.ui-toc-dropdown .nav>li>a:focus,.ui-toc-dropdown .nav>li>a:hover{background-color:initial;border-left:1px solid #000;color:#000;padding-left:19px;text-decoration:none}.ui-toc-dropdown[dir=rtl] .nav>li>a:focus,.ui-toc-dropdown[dir=rtl] .nav>li>a:hover{border-left:none;border-right:1px solid #000;padding-right:19px}.ui-toc-dropdown .nav>.active:focus>a,.ui-toc-dropdown .nav>.active:hover>a,.ui-toc-dropdown .nav>.active>a{background-color:initial;border-left:2px solid #000;color:#000;font-weight:700;padding-left:18px}.ui-toc-dropdown[dir=rtl] .nav>.active:focus>a,.ui-toc-dropdown[dir=rtl] .nav>.active:hover>a,.ui-toc-dropdown[dir=rtl] .nav>.active>a{border-left:none;border-right:2px solid #000;padding-right:18px}.ui-toc-dropdown .nav .nav{display:none;padding-bottom:10px}.ui-toc-dropdown .nav>.active>ul{display:block}.ui-toc-dropdown .nav .nav>li>a{font-size:12px;font-weight:400;padding-bottom:1px;padding-left:30px;padding-top:1px}.ui-toc-dropdown[dir=rtl] .nav .nav>li>a{padding-right:30px}.ui-toc-dropdown .nav .nav>li>ul>li>a{font-size:12px;font-weight:400;padding-bottom:1px;padding-left:40px;padding-top:1px}.ui-toc-dropdown[dir=rtl] .nav .nav>li>ul>li>a{padding-right:40px}.ui-toc-dropdown .nav .nav>li>a:focus,.ui-toc-dropdown .nav .nav>li>a:hover{padding-left:29px}.ui-toc-dropdown[dir=rtl] .nav .nav>li>a:focus,.ui-toc-dropdown[dir=rtl] .nav .nav>li>a:hover{padding-right:29px}.ui-toc-dropdown .nav .nav>li>ul>li>a:focus,.ui-toc-dropdown .nav .nav>li>ul>li>a:hover{padding-left:39px}.ui-toc-dropdown[dir=rtl] .nav .nav>li>ul>li>a:focus,.ui-toc-dropdown[dir=rtl] .nav .nav>li>ul>li>a:hover{padding-right:39px}.ui-toc-dropdown .nav .nav>.active:focus>a,.ui-toc-dropdown .nav .nav>.active:hover>a,.ui-toc-dropdown .nav .nav>.active>a{font-weight:500;padding-left:28px}.ui-toc-dropdown[dir=rtl] .nav .nav>.active:focus>a,.ui-toc-dropdown[dir=rtl] .nav .nav>.active:hover>a,.ui-toc-dropdown[dir=rtl] .nav .nav>.active>a{padding-right:28px}.ui-toc-dropdown .nav .nav>.active>.nav>.active:focus>a,.ui-toc-dropdown .nav .nav>.active>.nav>.active:hover>a,.ui-toc-dropdown .nav .nav>.active>.nav>.active>a{font-weight:500;padding-left:38px}.ui-toc-dropdown[dir=rtl] .nav .nav>.active>.nav>.active:focus>a,.ui-toc-dropdown[dir=rtl] .nav .nav>.active>.nav>.active:hover>a,.ui-toc-dropdown[dir=rtl] .nav .nav>.active>.nav>.active>a{padding-right:38px}.markdown-body{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica Neue,Helvetica,Roboto,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol}html[lang^=ja] .markdown-body{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica Neue,Helvetica,Roboto,Arial,Hiragino Kaku Gothic Pro,ヒラギノ角ゴ Pro W3,Osaka,Meiryo,メイリオ,MS Gothic,ＭＳ ゴシック,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol}html[lang=zh-tw] .markdown-body{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica Neue,Helvetica,Roboto,Arial,PingFang TC,Microsoft JhengHei,微軟正黑,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol}html[lang=zh-cn] .markdown-body{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica Neue,Helvetica,Roboto,Arial,PingFang SC,Microsoft YaHei,微软雅黑,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol}html .markdown-body[lang^=ja]{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica Neue,Helvetica,Roboto,Arial,Hiragino Kaku Gothic Pro,ヒラギノ角ゴ Pro W3,Osaka,Meiryo,メイリオ,MS Gothic,ＭＳ ゴシック,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol}html .markdown-body[lang=zh-tw]{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica Neue,Helvetica,Roboto,Arial,PingFang TC,Microsoft JhengHei,微軟正黑,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol}html .markdown-body[lang=zh-cn]{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica Neue,Helvetica,Roboto,Arial,PingFang SC,Microsoft YaHei,微软雅黑,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol}html[lang^=ja] .ui-toc-dropdown{font-family:Source Sans Pro,Helvetica,Arial,Meiryo UI,MS PGothic,ＭＳ Ｐゴシック,sans-serif}html[lang=zh-tw] .ui-toc-dropdown{font-family:Source Sans Pro,Helvetica,Arial,Microsoft JhengHei UI,微軟正黑UI,sans-serif}html[lang=zh-cn] .ui-toc-dropdown{font-family:Source Sans Pro,Helvetica,Arial,Microsoft YaHei UI,微软雅黑UI,sans-serif}html .ui-toc-dropdown[lang^=ja]{font-family:Source Sans Pro,Helvetica,Arial,Meiryo UI,MS PGothic,ＭＳ Ｐゴシック,sans-serif}html .ui-toc-dropdown[lang=zh-tw]{font-family:Source Sans Pro,Helvetica,Arial,Microsoft JhengHei UI,微軟正黑UI,sans-serif}html .ui-toc-dropdown[lang=zh-cn]{font-family:Source Sans Pro,Helvetica,Arial,Microsoft YaHei UI,微软雅黑UI,sans-serif}.ui-affix-toc{max-height:70vh;max-width:15vw;overflow:auto;position:fixed;top:0}.back-to-top,.expand-toggle,.go-to-bottom{color:#999;display:block;font-size:12px;font-weight:500;margin-left:10px;margin-top:10px;padding:4px 10px}.back-to-top:focus,.back-to-top:hover,.expand-toggle:focus,.expand-toggle:hover,.go-to-bottom:focus,.go-to-bottom:hover{color:#563d7c;text-decoration:none}.back-to-top,.go-to-bottom{margin-top:0}.ui-user-icon{background-position:50%;background-repeat:no-repeat;background-size:cover;border-radius:50%;display:block;height:20px;margin-bottom:2px;margin-right:5px;margin-top:2px;width:20px}.ui-user-icon.small{display:inline-block;height:18px;margin:0 0 .2em;vertical-align:middle;width:18px}.ui-infobar>small>span{line-height:22px}.ui-infobar>small .dropdown{display:inline-block}.ui-infobar>small .dropdown a:focus,.ui-infobar>small .dropdown a:hover{text-decoration:none}.ui-more-info{color:#888;cursor:pointer;vertical-align:middle}.ui-more-info .fa{font-size:16px}.ui-connectedGithub,.ui-published-note{color:#888}.ui-connectedGithub{line-height:23px;white-space:nowrap}.ui-connectedGithub a.file-path{color:#888;padding-left:22px;text-decoration:none}.ui-connectedGithub a.file-path:active,.ui-connectedGithub a.file-path:hover{color:#888;text-decoration:underline}.ui-connectedGithub .fa{font-size:20px}.ui-published-note .fa{font-size:20px;vertical-align:top}.unselectable{-webkit-user-select:none;-o-user-select:none;user-select:none}.selectable{-webkit-user-select:text;-o-user-select:text;user-select:text}.inline-spoiler-section{cursor:pointer}.inline-spoiler-section .spoiler-text{background-color:#333;border-radius:2px}.inline-spoiler-section .spoiler-text>*{opacity:0}.inline-spoiler-section .spoiler-img{filter:blur(10px)}.inline-spoiler-section.raw{background-color:#333;border-radius:2px}.inline-spoiler-section.raw>*{opacity:0}.inline-spoiler-section.unveil{cursor:auto}.inline-spoiler-section.unveil .spoiler-text{background-color:#3333331a}.inline-spoiler-section.unveil .spoiler-text>*{opacity:1}.inline-spoiler-section.unveil .spoiler-img{filter:none}@media print{blockquote,div,img,pre,table{page-break-inside:avoid!important}a[href]:after{font-size:12px!important}}.markdown-body.slides{color:#222;position:relative;z-index:1}.markdown-body.slides:before{background-color:currentColor;bottom:0;box-shadow:0 0 0 50vw;content:"";display:block;left:0;position:absolute;right:0;top:0;z-index:-1}.markdown-body.slides section[data-markdown]{background-color:#fff;margin-bottom:1.5em;position:relative;text-align:center}.markdown-body.slides section[data-markdown] code{text-align:left}.markdown-body.slides section[data-markdown]:before{content:"";display:block;padding-bottom:56.23%}.markdown-body.slides section[data-markdown]>div:first-child{left:1em;max-height:100%;overflow:hidden;position:absolute;right:1em;top:50%;transform:translateY(-50%)}.markdown-body.slides section[data-markdown]>ul{display:inline-block}.markdown-body.slides>section>section+section:after{border:3px solid #777;content:"";height:1.5em;position:absolute;right:1em;top:-1.5em}.site-ui-font{font-family:Source Sans Pro,Helvetica,Arial,sans-serif}html[lang^=ja] .site-ui-font{font-family:Source Sans Pro,Helvetica,Arial,Hiragino Kaku Gothic Pro,ヒラギノ角ゴ Pro W3,Osaka,Meiryo,メイリオ,MS Gothic,ＭＳ ゴシック,sans-serif}html[lang=zh-tw] .site-ui-font{font-family:Source Sans Pro,Helvetica,Arial,PingFang TC,Microsoft JhengHei,微軟正黑,sans-serif}html[lang=zh-cn] .site-ui-font{font-family:Source Sans Pro,Helvetica,Arial,PingFang SC,Microsoft YaHei,微软雅黑,sans-serif}body{font-smoothing:subpixel-antialiased!important;-webkit-font-smoothing:subpixel-antialiased!important;-moz-osx-font-smoothing:auto!important;-webkit-overflow-scrolling:touch;font-family:Source Sans Pro,Helvetica,Arial,sans-serif;letter-spacing:.025em}html[lang^=ja] body{font-family:Source Sans Pro,Helvetica,Arial,Hiragino Kaku Gothic Pro,ヒラギノ角ゴ Pro W3,Osaka,Meiryo,メイリオ,MS Gothic,ＭＳ ゴシック,sans-serif}html[lang=zh-tw] body{font-family:Source Sans Pro,Helvetica,Arial,PingFang TC,Microsoft JhengHei,微軟正黑,sans-serif}html[lang=zh-cn] body{font-family:Source Sans Pro,Helvetica,Arial,PingFang SC,Microsoft YaHei,微软雅黑,sans-serif}abbr[title]{border-bottom:none;text-decoration:underline;-webkit-text-decoration:underline dotted;text-decoration:underline dotted}abbr[data-original-title],abbr[title]{cursor:help}body.modal-open{overflow-y:auto;padding-right:0!important}svg{text-shadow:none}
    </style>
    <!-- HTML5 shim and Respond.js for IE8 support of HTML5 elements and media queries -->
    <!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
    <!--[if lt IE 9]>
    	<script src="https://cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv.min.js" integrity="sha256-3Jy/GbSLrg0o9y5Z5n1uw0qxZECH7C6OQpVBgNFYa0g=" crossorigin="anonymous"></script>
    	<script src="https://cdnjs.cloudflare.com/ajax/libs/respond.js/1.4.2/respond.min.js" integrity="sha256-g6iAfvZp+nDQ2TdTR/VVKJf3bGro4ub5fvWSWVRi2NE=" crossorigin="anonymous"></script>
		<script src="https://cdnjs.cloudflare.com/ajax/libs/es5-shim/4.5.9/es5-shim.min.js" integrity="sha256-8E4Is26QH0bD52WoQpcB+R/tcWQtpzlCojrybUd7Mxo=" crossorigin="anonymous"></script>
    <![endif]-->
</head>

<body>
    <div id="doc" class="markdown-body container-fluid comment-enabled" data-hard-breaks="true"><h1 style="color: Brown" id="Genes-of-Mice-and-Men" data-id="Genes-of-Mice-and-Men"><a class="anchor hidden-xs" href="#Genes-of-Mice-and-Men" title="Genes-of-Mice-and-Men"><span class="octicon octicon-link"></span></a><center>Genes of Mice and Men</center></h1><p><img src="https://i.imgur.com/ML1UCWo.png" alt="" loading="lazy"></p><p><span class="toc"><ul>
<li><a href="#Genes-of-Mice-and-Men" title="Genes of Mice and Men">Genes of Mice and Men</a><ul>
<li><a href="#Background" title="Background">Background</a><ul>
<li><a href="#Research-Question" title="Research Question:">Research Question:</a></li>
</ul>
</li>
<li><a href="#Procedure" title="Procedure">Procedure</a><ul>
<li><a href="#Bash" title="Bash">Bash</a></li>
<li><a href="#Python" title="Python">Python</a></li>
<li><a href="#RStudio" title="RStudio">RStudio</a></li>
</ul>
</li>
<li><a href="#ResultsData-Utilization" title="Results/Data Utilization">Results/Data Utilization</a></li>
<li><a href="#Discussion" title="Discussion">Discussion</a></li>
<li><a href="#References" title="References">References</a></li>
</ul>
</li>
</ul>
</span></p><h2 style="color: Brown" id="Background" data-id="Background"><a class="anchor hidden-xs" href="#Background" title="Background"><span class="octicon octicon-link"></span></a><center>Background</center></h2><h3 id="Research-Question" data-id="Research-Question"><a class="anchor hidden-xs" href="#Research-Question" title="Research-Question"><span class="octicon octicon-link"></span></a><span>Research Question:</span></h3><p><u><span>Which genes are conserved between </span><i><span>Homo sapiens</span></i><span> and </span><i><span>Mus musculus</span></i><span> and can be used to study human disease? Specifically, which genes can we study in a mouse model to research human cerebellum function and disease?</span></u></p><p><span>Mice and humans share approximately 70% of the same protein-coding sequences (with protein coding sequences making up approximately 1.5% of each respective 3 billion base pair genome). This, along with fast reproduction, ease of care, and having short life spans, make mice a powerful test subject to use for furthering our understanding of complex human genetic diseases. Therefore, in order to maintain the progress made in research in gene annotation and novel drug discovery it is vital to find orthologous regions of DNA between mice and humans. Exploring differences between gene transcription across species may also help us to understand disease formation and mechanisms.[1][2]</span></p><p><span>Orthologous genes are the result of genes conserved between two species after a divergent event from a common ancestor. According to Nature[3] only 1% of human genes do not have a mouse ortholog, and between those orthologous genes is an average sequence similarity of 78.5%. If the goal is to utilize the mouse genome to make novel discoveries in realizing how a certain gene is playing its part in disease/function, it would be beneficial to know which genes share the highest sequence and tissue expression similarity.</span></p><p><span>One of the anatomical areas that can be researched between humans and mice is the brain, more specifically the cerebellum, not only due to the aforementioned genetic similarities but also the anatomical resemblance. The cerebellum is located at the back of the head near where the spinal cord connects, and helps to coordinate and regulate a wide range of functions and processes in both your brain and body. [8] A disorder in the cerebellum can have effects such as; types of ataxia (failure of muscle control), brain degeneration, and cancer. The answer to what causes the development of such diseases/function may one day be found on the microscopic level of DNA, but first we must find the orthologs.</span></p><h2 style="color: Brown" id="Procedure" data-id="Procedure"><a class="anchor hidden-xs" href="#Procedure" title="Procedure"><span class="octicon octicon-link"></span></a><center>Procedure</center></h2><p><span>We took orthologous gene data from our own </span><strong><span>BlastP</span></strong><span> data table, Mouse and human brain transcription datatables from the </span><strong><span>Human Protein Atlas</span></strong><span> database. We then merged the three datasets by common gene name.</span></p><p><img src="https://i.imgur.com/PhBwMry.png" alt="" loading="lazy"></p><h3 style="color: Orange" id="Bash" data-id="Bash"><a class="anchor hidden-xs" href="#Bash" title="Bash"><span class="octicon octicon-link"></span></a><center>Bash</center></h3><ol>
<li><span>Start by downloading the Mouse and Human proteomes from Uniprot.</span><br>
<span>(Link below)</span></li>
<li><span>Using “mv” or “cp” command, move these downloads into a workable directory.</span></li>
<li><span>use gunzip to unzip the compressed fasta files</span></li>
<li><span>using blast+ program use the “makeblastdb” function and add in the unzipped human proteome</span></li>
<li><span>The mouse proteome is formatted the same as a multiFASTA file so we can use it as our query.</span></li>
<li><span>Using standard blast command line:</span></li>
</ol><pre><code>blastp -query mouse_proteome.fasta -db human_proteome.fasta \
    -out mouse_vs_human_blast_results.tab \
    -evalue 1e-20 \
    -outfmt 6\
    -max_target_seqs 1
</code></pre><p><span>Using -outfmt 6 we can get a much simpler data table that does not include all the additional comments like query number and number of hits.</span></p><p><span>Also using max_target_seqs 1 we reduce the amount of returned protein matches to only 1.</span></p><h3 style="color: Orange" id="Python" data-id="Python"><a class="anchor hidden-xs" href="#Python" title="Python"><span class="octicon octicon-link"></span></a><center>Python</center></h3><ol start="7">
<li><span>From this point we have a BLAST file but wanted only the gene names of the human gene and mouse gene. So using Python and the Regular Expression (re) module we were able to extract the names only and create our first data table.</span></li>
</ol><p><img src="https://i.imgur.com/8O52Erm.png" alt="" loading="lazy"></p><pre><code>import re
results_location = "/home/will/Proj_2/mouse_vs_human_blast_results.tab"

results = open(results_location, "r")
out_list = ["Mouse Gene, Human Gene"]
for line in results:
    splits = line.split("\t")
    mouse = (splits[0])
    subgroups_mouse = re.search("((\w*)$)", mouse)
    mouse = (subgroups_mouse.group(0))
    human = (splits[1])
    subgroups_human = re.search("((\w*)$)", human)
    human = (subgroups_human.group(0))
    addition = "\n"+mouse+","+human
    out_list.append(addition)

outfile = open("/home/will/Proj_2/Mouse_Human.csv", "w")

for i in out_list:
    outfile.write(i)

outfile.close(
</code></pre><p><img src="https://i.imgur.com/7RMM29G.png" alt="" loading="lazy"></p><p><span>*using the command: “blast_Mus_v_humo &lt;- unique(Mus_v_humo)” we removed repition introduced by a problem created with the blastp argument, “max_target_seqs 1”</span></p><h3 style="color: Orange" id="RStudio" data-id="RStudio"><a class="anchor hidden-xs" href="#RStudio" title="RStudio"><span class="octicon octicon-link"></span></a><center>RStudio</center></h3><ol start="8">
<li>
<p><span>The second table we used was downloaded from ProteinAtlas. It contained human genes and tissue expression data expressed in nTPM (normalized Transcript Per Million) for humans. After removing a couple of unwanted data columns we were left with the following table:</span><br>
<img src="https://i.imgur.com/Bg99xAk.png" alt="" loading="lazy"></p>
</li>
<li>
<p><span>Using a column filter we removed all rows that did not include the “cerebellum” tissue. Now we had a table with nTPM readings only in the cerebellum and could merge it with our BLAST result table.</span><br>
<img src="https://i.imgur.com/MS02BGk.png" alt="" loading="lazy"></p>
</li>
<li>
<p><span>Using R, we merged the Blast results using “Human_Gene” and the human expression table “Gene_name.” This returned the first merged table but to make meaningful comparisons between mice and human cerebllum genes we needed tissue level mouse RNA-seq data.</span></p>
<pre><code>merge_1 &lt;- left_join(blast_Mus_v_humo, human_cereb, by=c("Human_gene"= "Gene_name"))
</code></pre>
</li>
<li>
<p><span>The table populated for all the human cerebellum genes and this is where our first level of tissue specificity is introduced. Many genes found using BLASTp were not expressed in the cerebellum so we removed their rows. Using two commands we removed genes that had no cerebellum (or very little &lt;1) nTPM reads.</span></p>
<pre><code>merge_1 %&gt;% drop_na("Human_ENSG") 

merge_1 &lt;- filter(merge_1, Human_nTPM &gt; 1)
</code></pre>
</li>
<li>
<p><span>Now using mouse data from Protein Atlas we filtered the table again for nTPM in the cerebllum and Mouse Gene name. The original data table was over 100 columns long but we used colnames to reduce this drastically. We included the Gene description and other general functional annotations for a more robust final data table.</span></p>
</li>
</ol><p><img src="https://i.imgur.com/OAkr3Zm.png" alt="" loading="lazy"></p><ol start="13">
<li>
<p><span>Now using leftjoin we merged the mouse data table with our first merge table. We merged the mouse table onto the other table aligning on Mouse_gene and Gene. We then renamed the nTPM columns to be specific by species and moved them next to one another.</span></p>
<pre><code>merge_2 &lt;- merge_2 %&gt;% relocate(12, .after = 4)
</code></pre>
</li>
<li>
<p><span>Using the method from earlier we removed any N/A column that was a Mouse_gene with no associated annotation from the mouse table. What we noticed after this merge is that every mouse gene and human gene had the same gene name. This was very interesting because it showed us how genome annotations are created. Most genomes are referenced back to the human genome and named accordingly. Removing these N/A columns along with again filtering against genes with less than 1 nTPM returned our final table for species-species cerebellum specific gene expression table.</span></p>
</li>
</ol><p><img src="https://i.imgur.com/NFJ4R6S.png" alt="" loading="lazy"></p><p><span>*Disease involvement is a column to the right.</span></p><pre><code>setwd("C:/Users/Will/Desktop")

library(tidyverse)
library(dplyr)

Mus_v_humo &lt;- read_csv(file = "Mouse_Human_clean.csv")

#view(Mus_v_humo)

mice &lt;- read_tsv(file = "mouse_brain_category_rna_cerebral.tsv")

mouse_rough &lt;- mice

#view(mouse_rough)

humans &lt;- read_tsv(file = "rna_tissue_gtex.tsv")

humans = humans[-c(4,5)]

human_cereb &lt;- filter(humans, Tissue == "cerebellum")

colnames(human_cereb)[4] = "Human_nTPM"

human_cereb &lt;- human_cereb[-c(3)]

colnames(human_cereb)[1:2] = c("Human_ENSG", "Gene_name")

#view(human_cereb)

#view(human_cereb)

#view(Mus_v_humo)

blast_Mus_v_humo &lt;- unique(Mus_v_humo)

colnames(blast_Mus_v_humo) = c("Mouse_gene", "Human_gene")

#view(blast_Mus_v_humo)

mice &lt;- mice[c(1:17,93)]

mice_awk &lt;- mice[-c(2,6,7,12:16)]

colnames(mice_awk)[5] = "Protein_desc"

view(mice_awk)

merge_1 &lt;- left_join(blast_Mus_v_humo, human_cereb, by=c("Human_gene"= "Gene_name"))

merge_1 %&gt;% drop_na("Human_ENSG") 

merge_1 &lt;- filter(merge_1, Human_nTPM &gt; 1)

view(merge_1)

merge_2 &lt;- left_join(merge_1, mice_awk, by=c("Mouse_gene"= "Gene"))

merge_2 &lt;- merge_2 %&gt;% relocate(13, .after = 4)

merge_2 &lt;- merge_2 %&gt;% drop_na("Protein_desc") 

colnames(merge_2)[5] = "Mouse_nTPM"

colnames(merge_2)[13] = "RNA_tissue_specificity"

merge_2 &lt;- merge_2[-c(3,6,8)]

view(merge_2)

write_csv( merge_2, file = "Final_table.csv")
</code></pre><p><span>An optional filtering to remove genes expressed throughout all tissues can be used to further refine the dataset to genes that may be more specific to the cerebllum.</span></p><pre><code>    merge_2 &lt;- filter(merge_2, Mouse_nTPM &gt; 1)

    merge_2 &lt;- filter(merge_2, RNA_tissue_specificity != "Detected in all")
</code></pre><h2 style="color: Brown" id="ResultsData-Utilization" data-id="ResultsData-Utilization"><a class="anchor hidden-xs" href="#ResultsData-Utilization" title="ResultsData-Utilization"><span class="octicon octicon-link"></span></a><center>Results/Data Utilization</center></h2><p><img src="https://i.imgur.com/t7cG6fN.png" alt="" loading="lazy"></p><p><span>*Zoomed in Final data table.</span></p><p><span>The most applicable genes that could be used for research with mice would be the ones that show the highest sequence similarity and species to species expression. We have (1) identified orthologous proteins between mice and humans and (2) applied normalized transcription data for each gene from each respective species. Our data table holds over 3000 unique genes along with nTPM data between species and their respective annotations.</span></p><p><span>A researcher can look at our table and learn about the cerebellar function of genes such as: biological process, disease involvement, and molecular function and decide which genes to make perturbations to research a disease of interest. A quick analysis that can also be performed is to find which of the genes share a similar expression level to each other. Here is how we used our table to find genes whose transcription levels were within a user specified % similarity to each other using Python:</span></p><pre><code>results_location = "/[users director]/Final_table.csv"

results = open(results_location, "r")

header = results.readline()

# input your tolerance limit as a percentage difference

tolerance = input("Enter you percentage difference in similarity as a number:")

if tolerance.isnumeric():
    tolerance = float(tolerance)
    Similar_expression = []
    sp_sp_expression = open("/[user's directory]/Similar_spx2_" + str(tolerance) + ".csv", "w")
    sp_sp_expression.write(header)
    for line in results:
        splits = line.split(",")
        # [3] == human nTPM
        #print(splits[3], splits[4])
        human_nTPM = float(splits[3])
        mice_nPTM = float(splits[4])
        first_step = abs(human_nTPM-mice_nPTM)
        second_step = (mice_nPTM+human_nTPM)/2
        perc_diff = (first_step/second_step)*100
        if perc_diff &lt; tolerance:
            sp_sp_expression.write(line)
        else:
            continue
    sp_sp_expression.close()
else:
    exit("Invalid input, Exiting program")
</code></pre><p><span>Another way that this table can be utilized is to make phylogenetic inferences on the evolution from the mouse cerebellum to the human cerebellum. Significant differences in expression of the same gene shows how the different species have diverged by illustrating how different proteins became more/less important for cerebellar processes for each respective species. We can also infer that genes with similar transcription levels between species are most likely more conserved in function.</span></p><h2 style="color: Brown" id="Discussion" data-id="Discussion"><a class="anchor hidden-xs" href="#Discussion" title="Discussion"><span class="octicon octicon-link"></span></a><center>Discussion</center></h2><p><span>We successfully merged transcriptional data across species to compare shared genes between humans and mice. We thought the utilization of BlastP to find orthologous genes would enable us to link together gene names that were not the same, however we found that all of the orthologous genes that had expression in cerebellum tissue already shared  a common gene name. We learned that the model used for creating gene names in novel genomes is heavily referenced back to the human genome. Which allows for easy comparisons. While the names of these genes were shared, our datatable shows that these genes were sometimes transcribed at significantly different levels between species. Our postulation is that the most fitting genes that can be used in studying human disease and function would be those that have similar transcription levels and gene interactions. Take for example the gene NOP56:</span></p><p><span>According to our table, NOP56 is a ribonucleoprotein that is involved with neurodegeneration and spinocerebellar ataxia. This gene showed similar cerebellum tissue expression level across species (Humans: 47nTPM, Mice: 39nTPM). In this way NOP56 could be utilized in research for these aforementioned diseases, but the interactions that this gene has with other genes will be needed to take into account to truly study the disease as a whole. If researchers wish to study NOP56 in humans using mice, they need to understand that the gene interactions and transcription levels could be different so they must adjust their research methods accordingly.</span></p><p><strong><span>Human NOP56 Gene Network:</span></strong><br>
<img src="https://i.imgur.com/JrfSiVD.png" alt="" loading="lazy"></p><p><strong><span>Mice NOP56 Gene Network:</span></strong><br>
<img src="https://i.imgur.com/g8UhAT3.png" alt="" loading="lazy"></p><p><strong><span>Shared:</span></strong></p><ul>
<li><span>NOP58</span></li>
<li><span>DKC1</span></li>
<li><span>RRP9</span></li>
<li><span>FBL</span></li>
<li><span>NHP2L1</span></li>
<li><span>NHP2</span></li>
</ul><p><strong><span>Different:</span></strong></p><ul>
<li><span>MPHOSPH10 - human</span></li>
<li><span>HEATR1 - human</span></li>
<li><span>KRR1 - human</span></li>
<li><span>WDR36 - human</span></li>
<li><span>WDR43 - mouse</span></li>
<li><span>GM5616 - mouse</span></li>
<li><span>FTSJ3 - mouse</span></li>
<li><span>NIP7 - mouse</span></li>
</ul><p><span>Moving into the future we would like to build this into a more robust database. A database that connects not just humans to mice but humans to other model organisms to guide disease research to delineate the best model organism. The database would be very reliant on RNA-Seq data and would be limited in scope due to the scarcity of this information source. We would also like to automate our process to create a database of similar tables that take into account user interest to produce a more specific/meaningful table. We can also expand the database past the cerebellum specific genes and have categories for every tissue or mutliple tissue genes.</span></p><h2 id="References" data-id="References"><a class="anchor hidden-xs" href="#References" title="References"><span class="octicon octicon-link"></span></a><span>References</span></h2><p><span>[1]Why Mouse matters</span><br>
<a href="https://www.genome.gov/10001345/importance-of-mouse-genome#:~:text=On%20average%2C%20the%20protein%2Dcoding,they%20are%20required%20for%20function" target="_blank" rel="noopener"><span>https://www.genome.gov/10001345/importance-of-mouse-genome#:~:text=On average%2C the protein-coding,they are required for function</span></a><span>.</span></p><p><span>[2]Why use the mouse in research</span><br>
<a href="https://www.yourgenome.org/facts/why-use-the-mouse-in-research/" target="_blank" rel="noopener"><span>https://www.yourgenome.org/facts/why-use-the-mouse-in-research/</span></a></p><p><span>[3]Humanising the mouse genome</span><br>
<a href="https://www.nature.com/articles/s41467-019-09716-7#:~:text=Approximately%201%25%20of%20human%20genes,numbers%20in%20the%20two%20species" target="_blank" rel="noopener"><span>https://www.nature.com/articles/s41467-019-09716-7#:~:text=Approximately 1%25 of human genes,numbers in the two species</span></a><span>.</span></p><p><span>[4]Protein ID conversion tool - create name table</span><br>
<a href="https://www.genenames.org/download/custom/" target="_blank" rel="noopener"><span>https://www.genenames.org/download/custom/</span></a></p><p><span>[5]Mouse Proteome:</span><br>
<a href="https://www.uniprot.org/proteomes/UP000000589" target="_blank" rel="noopener"><span>https://www.uniprot.org/proteomes/UP000000589</span></a></p><p><span>[6]Human Proteome:</span><br>
<a href="https://www.uniprot.org/proteomes/UP000005640" target="_blank" rel="noopener"><span>https://www.uniprot.org/proteomes/UP000005640</span></a></p><p><span>[7]A comparison of human and mouse gene co-expression networks reveals conservation and divergence at the tissue, pathway and disease levels</span><br>
<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4654840/" target="_blank" rel="noopener"><span>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4654840/</span></a></p><p><span>[8]What is the cerebellum</span><br>
<a href="https://my.clevelandclinic.org/health/body/23418-cerebellum" target="_blank" rel="noopener"><span>https://my.clevelandclinic.org/health/body/23418-cerebellum</span></a></p><p><span>[9]Cerebellar Disorders</span><br>
<a href="https://medlineplus.gov/cerebellardisorders.html" target="_blank" rel="noopener"><span>https://medlineplus.gov/cerebellardisorders.html</span></a></p><p><span>[10]Website Page containing our used databases</span><br>
<a href="https://www.proteinatlas.org/about/download" target="_blank" rel="noopener"><span>https://www.proteinatlas.org/about/download</span></a><br>
<span>Human [RNA GTEx brain region gene data]</span><br>
<span>Mice [RNA mouse brain region gene data]</span></p></div>
    <div class="ui-toc dropup unselectable hidden-print" style="display:none;">
        <div class="pull-right dropdown">
            <a id="tocLabel" class="ui-toc-label btn btn-default" data-toggle="dropdown" href="#" role="button" aria-haspopup="true" aria-expanded="false" title="Table of content">
                <i class="fa fa-bars"></i>
            </a>
            <ul id="ui-toc" class="ui-toc-dropdown dropdown-menu" aria-labelledby="tocLabel">
                <div class="toc"><ul class="nav">
<li><a href="#Genes-of-Mice-and-Men" title="Genes of Mice and Men">Genes of Mice and Men</a><ul class="nav">
<li><a href="#Background" title="Background">Background</a><ul class="nav">
<li><a href="#Research-Question" title="Research Question:">Research Question:</a></li>
</ul>
</li>
<li><a href="#Procedure" title="Procedure">Procedure</a><ul class="nav">
<li><a href="#Bash" title="Bash">Bash</a></li>
<li><a href="#Python" title="Python">Python</a></li>
<li><a href="#RStudio" title="RStudio">RStudio</a></li>
</ul>
</li>
<li><a href="#ResultsData-Utilization" title="Results/Data Utilization">Results/Data Utilization</a></li>
<li><a href="#Discussion" title="Discussion">Discussion</a></li>
<li><a href="#References" title="References">References</a></li>
</ul>
</li>
</ul>
</div><div class="toc-menu"><a class="expand-toggle" href="#">Expand all</a><a class="back-to-top" href="#">Back to top</a><a class="go-to-bottom" href="#">Go to bottom</a></div>
            </ul>
        </div>
    </div>
    <div id="ui-toc-affix" class="ui-affix-toc ui-toc-dropdown unselectable hidden-print" data-spy="affix" style="top:17px;display:none;" null null>
        <div class="toc"><ul class="nav">
<li><a href="#Genes-of-Mice-and-Men" title="Genes of Mice and Men">Genes of Mice and Men</a><ul class="nav">
<li><a href="#Background" title="Background">Background</a><ul class="nav">
<li><a href="#Research-Question" title="Research Question:">Research Question:</a></li>
</ul>
</li>
<li><a href="#Procedure" title="Procedure">Procedure</a><ul class="nav">
<li><a href="#Bash" title="Bash">Bash</a></li>
<li><a href="#Python" title="Python">Python</a></li>
<li><a href="#RStudio" title="RStudio">RStudio</a></li>
</ul>
</li>
<li><a href="#ResultsData-Utilization" title="Results/Data Utilization">Results/Data Utilization</a></li>
<li><a href="#Discussion" title="Discussion">Discussion</a></li>
<li><a href="#References" title="References">References</a></li>
</ul>
</li>
</ul>
</div><div class="toc-menu"><a class="expand-toggle" href="#">Expand all</a><a class="back-to-top" href="#">Back to top</a><a class="go-to-bottom" href="#">Go to bottom</a></div>
    </div>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.1.1/jquery.min.js" integrity="sha256-hVVnYaiADRTO2PzUGmuLJr8BLUSjGIZsDYGmIJLv2b8=" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous" defer></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/gist-embed/2.6.0/gist-embed.min.js" integrity="sha256-KyF2D6xPIJUW5sUDSs93vWyZm+1RzIpKCexxElmxl8g=" crossorigin="anonymous" defer></script>
    <script>
        var markdown = $(".markdown-body");
        //smooth all hash trigger scrolling
        function smoothHashScroll() {
            var hashElements = $("a[href^='#']").toArray();
            for (var i = 0; i < hashElements.length; i++) {
                var element = hashElements[i];
                var $element = $(element);
                var hash = element.hash;
                if (hash) {
                    $element.on('click', function (e) {
                        // store hash
                        var hash = this.hash;
                        if ($(hash).length <= 0) return;
                        // prevent default anchor click behavior
                        e.preventDefault();
                        // animate
                        $('body, html').stop(true, true).animate({
                            scrollTop: $(hash).offset().top
                        }, 100, "linear", function () {
                            // when done, add hash to url
                            // (default click behaviour)
                            window.location.hash = hash;
                        });
                    });
                }
            }
        }

        smoothHashScroll();
        var toc = $('.ui-toc');
        var tocAffix = $('.ui-affix-toc');
        var tocDropdown = $('.ui-toc-dropdown');
        //toc
        tocDropdown.click(function (e) {
            e.stopPropagation();
        });

        var enoughForAffixToc = true;

        function generateScrollspy() {
            $(document.body).scrollspy({
                target: ''
            });
            $(document.body).scrollspy('refresh');
            if (enoughForAffixToc) {
                toc.hide();
                tocAffix.show();
            } else {
                tocAffix.hide();
                toc.show();
            }
            $(document.body).scroll();
        }

        function windowResize() {
            //toc right
            var paddingRight = parseFloat(markdown.css('padding-right'));
            var right = ($(window).width() - (markdown.offset().left + markdown.outerWidth() - paddingRight));
            toc.css('right', right + 'px');
            //affix toc left
            var newbool;
            var rightMargin = (markdown.parent().outerWidth() - markdown.outerWidth()) / 2;
            //for ipad or wider device
            if (rightMargin >= 133) {
                newbool = true;
                var affixLeftMargin = (tocAffix.outerWidth() - tocAffix.width()) / 2;
                var left = markdown.offset().left + markdown.outerWidth() - affixLeftMargin;
                tocAffix.css('left', left + 'px');
            } else {
                newbool = false;
            }
            if (newbool != enoughForAffixToc) {
                enoughForAffixToc = newbool;
                generateScrollspy();
            }
        }
        $(window).resize(function () {
            windowResize();
        });
        $(document).ready(function () {
            windowResize();
            generateScrollspy();
        });

        //remove hash
        function removeHash() {
            window.location.hash = '';
        }

        var backtotop = $('.back-to-top');
        var gotobottom = $('.go-to-bottom');

        backtotop.click(function (e) {
            e.preventDefault();
            e.stopPropagation();
            if (scrollToTop)
                scrollToTop();
            removeHash();
        });
        gotobottom.click(function (e) {
            e.preventDefault();
            e.stopPropagation();
            if (scrollToBottom)
                scrollToBottom();
            removeHash();
        });

        var toggle = $('.expand-toggle');
        var tocExpand = false;

        checkExpandToggle();
        toggle.click(function (e) {
            e.preventDefault();
            e.stopPropagation();
            tocExpand = !tocExpand;
            checkExpandToggle();
        })

        function checkExpandToggle () {
            var toc = $('.ui-toc-dropdown .toc');
            var toggle = $('.expand-toggle');
            if (!tocExpand) {
                toc.removeClass('expand');
                toggle.text('Expand all');
            } else {
                toc.addClass('expand');
                toggle.text('Collapse all');
            }
        }

        function scrollToTop() {
            $('body, html').stop(true, true).animate({
                scrollTop: 0
            }, 100, "linear");
        }

        function scrollToBottom() {
            $('body, html').stop(true, true).animate({
                scrollTop: $(document.body)[0].scrollHeight
            }, 100, "linear");
        }
    </script>
</body>

</html>
